prov/efa/src/rxr/rxr_read.c

```diff
@@ -129,47 +165,20 @@ struct rxr_read_entry *rxr_read_alloc_entry(struct rxr_ep *ep, int entry_type, v
 		read_entry->rma_iov_count = rx_entry->rma_iov_count;
 		read_entry->rma_iov = rx_entry->rma_iov;
 
-		mr_desc = rx_entry->desc;
 		total_iov_len = ofi_total_iov_len(rx_entry->iov, rx_entry->iov_count);
 		total_rma_iov_len = ofi_total_rma_iov_len(rx_entry->rma_iov, rx_entry->rma_iov_count);
 		read_entry->total_len = MIN(total_iov_len, total_rma_iov_len);
-	}
 
-	if (lower_ep_type == EFA_EP) {
-		/* EFA provider need local buffer registration */
-		for (i = 0; i < read_entry->iov_count; ++i) {
-			if (mr_desc && mr_desc[i]) {
-				read_entry->mr[i] = NULL;
-				read_entry->mr_desc[i] = mr_desc[i];
-			} else {
-				err = fi_mr_reg(rxr_ep_domain(ep)->rdm_domain,
-						read_entry->iov[i].iov_base, read_entry->iov[i].iov_len,
-						FI_RECV, 0, 0, 0, &read_entry->mr[i], NULL);
-
-				if (err == -FI_ENOMEM && efa_mr_cache_enable) {
```

Comment:
Before this change, registration happened during initialization and the retry is explicit.

This change move the registration to ```rxr_read_post```, and will return -FI_EAGAIN when registration failed. When progress engine get -FI_EAGAIN, it will retry, so we do not need to explicitly retry any more. I updated commit message to discuss this.